internal/acs/runner.go

```diff
@@ -0,0 +1,250 @@
+package acs
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/app-sre/acs-cve-prom-exporter/internal/collector"
+	"github.com/app-sre/acs-cve-prom-exporter/pkg/acsclient"
+	"github.com/app-sre/acs-cve-prom-exporter/pkg/worker"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type CollectorLoopParams struct {
+	Collector      *collector.CVECollector
+	Logger         log.Logger
+	Credentials    ACSCredentials
+	ExitSignal     chan struct{}
+	WorkerPoolSize int
+	Timeout        int
+	Retry          int
+}
+
+type ACSCredentials struct {
+	Endpoint string
+	Token    string
+}
+type cveSummaryData struct {
+	clusterName     string
+	namespaceName   string
+	instances       map[string]int
+	imageID         string
+	imageRegistry   string
+	imageRemote     string
+	imageTag        string
+	operatingSystem string
+	cveData         map[string]cveData
+}
+type cveData struct {
+	severity  string
+	cvss      float64
+	isFixable bool
+}
+
+func CVECollectorLoop(collectorParams CollectorLoopParams) {
+	acsClient, err := acsclient.NewACSClient(collectorParams.Credentials.Endpoint, collectorParams.Credentials.Token, collectorParams.Timeout, collectorParams.Retry)
+	if err != nil {
+		level.Error(collectorParams.Logger).Log("msg", "error creating acs client", "error", err.Error())
+		collectorParams.ExitSignal <- struct{}{}
+		return
+	}
+	for {
+		cveData, err := getCVEDataFromACS(acsClient, collectorParams.WorkerPoolSize)
+		if err != nil {
+			level.Error(collectorParams.Logger).Log("msg", "Error processing metrics::", "error", err.Error())
+			collectorParams.ExitSignal <- struct{}{}
+			return
+		}
+
+		collectorParams.Collector.MetricsData = getMetricDataFromCVE(cveData, collectorParams.Logger, collectorParams.Collector)
```

Comment:
do we need a lock here for `MetricsData`? since it's also used in collector, maybe need a lock with getter setter.